### ISSUE TITLE
Incorrect client code generation for explicitly implemented interface members

### ISSUE THREAD

**Issue by Inkp** [NSwag/NSwag#241]
Given following generation code:
```
        public static void Main()
        {
            var service = new SwaggerService
            {
                Host = "localhost",
                Schemes = new List<SwaggerSchema> { SwaggerSchema.Http },
                Produces = new List<string> { "application/json" },
            };
            service.Paths.Add("name", new SwaggerOperations
            {
                {
                    SwaggerOperationMethod.Get, new SwaggerOperation
                    {
                        OperationId = "id",
                        Consumes = new List<string> {"application/x-www-form-urlencoded"},
                        Parameters = new List<SwaggerParameter>()
                        {
                            new SwaggerParameter
                            {
                                Name = "body",
                                Kind = SwaggerParameterKind.Body,
                                IsRequired = true,
                                Schema = JsonSchema4.FromType(typeof(Foo))
                            }
                        }
                    }
                }
            });

            Console.WriteLine(new SwaggerToTypeScriptClientGenerator(service,
                new SwaggerToTypeScriptClientGeneratorSettings
                {
                    ClassName = "FooClient",
                    Template = TypeScriptTemplate.AngularJS
                }).GenerateFile());
        }

        public interface IFoo
        {
            string Prop { get; }
        }

        public class Foo : IFoo
        {
            string IFoo.Prop { get; }
        }
```
Generator yields following TS client which has erroneous member `swaggen.Program.IFoo.Prop`.
[test.txt](https://github.com/NSwag/NSwag/files/424451/test.txt)

**EDIT**: I wonder if that's an issue of NJsonSchema, since explicitly implemented properties are private and by my understanding should be excluded from schema

**Comment by rsuter** [comment]
Good find. Yes, DTO property handling is implemented in NJsonSchema... to check whether to include explicitly implemented properties or not, we have to check how Json.NET serializes the class... is the property included in the generated JSON data?

**Comment by Inkp** [comment]
Given following code:
```
public interface IFoo 
{
	string Prop { get; }
}

public class Foo : IFoo
{
	string IFoo.Prop => "Prop";
	private string Prep => "Prep";
	public string ThisShouldBeInTheOutput => "It is";
}
```
JSON.NET serializes `new Foo()` like this:
`{"ThisShouldBeInTheOutput":"It is"}`

While `JsonSchema4.FromType(typeof(Foo)).ToJson()` yields
```
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "x-typeName": "Foo",
  "additionalProperties": false,
  "properties": {
    "UserQuery.IFoo.Prop": {
      "type": [
        "null",
        "string"
      ]
    },
    "Prep": {
      "type": [
        "null",
        "string"
      ]
    },
    "ThisShouldBeInTheOutput": {
      "type": [
        "null",
        "string"
      ]
    }
  }
}
```

**Comment by rsuter** [comment]
Fixed in https://github.com/NJsonSchema/NJsonSchema/commit/6cfa3338ff9669a7692c3589bebda07455011c80

**Comment by Inkp** [comment]
Great, thanks for the fast fix! How soon will this be on the NuGet?

**Comment by rsuter** [comment]
Today or tomorrow...

**Comment by rsuter** [comment]
NSwag v5 has been released, please check if your problem is solved and reopen the issue if it still remains...

**Comment by rsuter** [comment]
Fixed in v5.2
